numba/cuda/cudadrv/devicearray.py

```diff
@@ -354,15 +354,37 @@ def view(self, dtype):
         copy of the data.
         """
         dtype = np.dtype(dtype)
-        if dtype.itemsize != self.dtype.itemsize:
-            raise TypeError("new dtype itemsize doesn't match")
+        shape = list(self.shape)
+        strides = list(self.strides)
+
+        if self.dtype.itemsize != dtype.itemsize:
+            if not self.is_c_contiguous():
+                raise ValueError(
+                    "To change to a dtype of a different size,"
+                    " the array must be C-contiguous"
+                )
```

Comment:
Is this restriction to `C-contiguous` necessary? Would a `dtype` change not also work fine on a `F-contiguous` array?

Reply:
NumPy deprecated that behavior a while ago. ( https://github.com/numpy/numpy/pull/6747 ) I suppose we could support it, but should we?

Reply:
hmmm, wasn't aware of that, I think that whilst `devicearray`s are relatively restricted, I'm reluctant to replicate something deprecated but seemingly useful, given there's a work-around that produces the same and doesn't break against NumPy behaviour. So basically, no, unless you feel otherwise?

Reply:
Sounds ok to me. So leave as-is or should we have a different error here?

Reply:
This is fine, thanks.